File: ExampleApp/MainWindow.cpp
#include "MainWindow.h"
#include "ExampleListItem.h"

#include <QtWidgets/qboxlayout.h>

namespace dak::ExampleApp
{
   using namespace QtAdditions;
   using namespace std;

   /////////////////////////////////////////////////////////////////////////
   //
   // Create the main window.

   MainWindow::MainWindow()
   {
      buildUI();
      fillUI();
      connectUI();
   }

   /////////////////////////////////////////////////////////////////////////
   //
   // Create the UI elements.

   void MainWindow::buildUI()
   {
      auto container = new QWidget;
      auto layout = new QHBoxLayout;
      container->setLayout(layout);

      _list1 = new QWidgetListWidget(nullptr, true, QBoxLayout::Direction::TopToBottom);
      //_list1 = new QWidgetListWidget(nullptr, true, QBoxLayout::Direction::LeftToRight);
      _list1->setAcceptDrops(true);
      _scrollList1 = new QWidgetScrollListWidget(_list1);
      layout->addWidget(_scrollList1);

      _list2 = new QWidgetListWidget(nullptr, true, QBoxLayout::Direction::TopToBottom);
      //_list2 = new QWidgetListWidget(nullptr, true, QBoxLayout::Direction::LeftToRight);
      _list2->setAcceptDrops(true);
      _scrollList2 = new QWidgetScrollListWidget(_list2);
      layout->addWidget(_scrollList2);

      setCentralWidget(container);
   }

   /////////////////////////////////////////////////////////////////////////
   //
   // Connect the signals of the UI elements.

   void MainWindow::connectUI()
   {
   }

   /////////////////////////////////////////////////////////////////////////
   //
   // Fill the UI with the intial data.

   void MainWindow::fillUI()
   {
      for (int i = 0; i < 10; ++i)
         _list1->addItem(new ExampleListItem(QString::asprintf("Item #%d", i+1)));
   }
}

File: PythonExample/mainWindow.py
from listItem import ListItem
from PySide2.QtWidgets import QHBoxLayout,QWidget, QMainWindow,QBoxLayout
from qWidgetListItem import QWidgetListItem
from qWidgetListWidget import QWidgetListWidget
from qWidgetScrollListWidget import QWidgetScrollListWidget

class MainWindow(QMainWindow):
    def __init__(self):
        super().__init__()
        scroll_list_1 = self.buildUI()
        self.fillUI(scroll_list_1.widget())
        self.connectUI()

    def build_scroll_list_widget(self, _dir = QBoxLayout.Direction.TopToBottom ):
        _list = QWidgetListWidget(stretch=True, _dir = _dir)
        #_list.setAcceptDrops(True)

        scroll_list = QWidgetScrollListWidget(_list)
        return scroll_list

    def buildUI(self):
        container = QWidget()
        layout = QHBoxLayout()

        scroll_list_1 = self.build_scroll_list_widget()
        scroll_list_2 = self.build_scroll_list_widget()

        layout.addWidget(scroll_list_1)
        layout.addWidget(scroll_list_2)

        container.setLayout(layout)
        self.setCentralWidget(container)

        return scroll_list_1

    def connectUI(self):
        pass

    def fillUI(self,_list):
        for i in range(10):
            _list.add_item(ListItem('Item {}'.format(i+1)))
